fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -2598,17 +2599,32 @@ public ParseNode visitCreateAnalyzeStatement(StarRocksParser.CreateAnalyzeStatem
             }
         }
 
+        StatsConstants.AnalyzeType analyzeType = StatsConstants.AnalyzeType.FULL;
+        if (context.FULL() != null) {
+            analyzeType = StatsConstants.AnalyzeType.FULL;
+        } else if (context.SAMPLE() != null) {
+            analyzeType = StatsConstants.AnalyzeType.SAMPLE;
+        } else if (context.histogramStatement() != null) {
+            analyzeType = StatsConstants.AnalyzeType.HISTOGRAM;
+        }
+        boolean isSample = analyzeType == StatsConstants.AnalyzeType.SAMPLE;
```

Comment:
should keep `context.FULL() == null`, not `analyzeType == StatsConstants.AnalyzeType.SAMPLE`